src/media/js/elements/select.js

```diff
@@ -0,0 +1,229 @@
+                // Value set in the createdCallback.
+                value: null,
+                writable: true
+            },
+            toggle: {
+                // Toggles visibility of the <mkt-option>s.
+                value: function(bool) {
+                    if (!this.getAttribute('data-mkt-select--aligned')) {
+                        if (this.alignOptions() > 0) {
+                            this.setAttribute('data-mkt-select--aligned', true);
+                        }
+                    }
+                    if (bool) {
+                        this.classList.toggle(VISIBLE, bool);
+                    } else {
+                        this.classList.toggle(VISIBLE);
+                    }
+                    return this;
+                },
+            },
+            alignOptions: {
+                value: function() {
+                    // Aligns options with the selected/visible top option.
+                    var offset = $('mkt-selected-text', this).position().left;
+                    $('mkt-option', this).css('padding-left', offset);
+                    return offset;
+                }
+            },
+            updateOptionIndices: {
+                value: function() {
+                    // Sets a data-index on each mkt-option that is not
+                    // [selected].  Prominently used to set colors since we
+                    // don't have CSS4 // nth-match.
+                    var selectedOption = this.querySelector(
+                        'mkt-option[selected]');
+                    var mktOptions = this.querySelectorAll('mkt-option');
+                    var index = 0;
+                    for (var i = 0; i < mktOptions.length; i++) {
+                        var mktOption = mktOptions[i];
+                        if (mktOption === selectedOption) {
+                            mktOption.removeAttribute(
+                                'data-mkt-option--index');
+                        } else {
+                            mktOption.setAttribute('data-mkt-option--index',
+                                                   index++);
+                        }
+                    }
+                }
+            }
+        }),
+    });
+
+
+    function copyAttrs(dest, source) {
+        // Copies attributes from one element to another.
+        for (var i = 0; i < source.attributes.length; i++) {
+            var attr = source.attributes[i];
+            if (attr !== 'id') {
+                dest.setAttribute(attr.name, attr.value);
+            }
+        }
+        return dest;
+    }
+
+    function proxyElement(htmlElement, customElement, key) {
+        // Proxy custom element interface to actual element.
+        Object.getOwnPropertyNames(htmlElement.prototype).forEach(function(method) {
+            try {
+                customElement.prototype[method] = {
+                    value: function() {
```

Comment:
I see what you're saying about prototypes. I thought this `{value: function() { ... }}` thing was for `Object.create` not for prototypes. I don't think prototypes are special? Should this just be `customElement.prototype[method] = function() { ... }`?